wandb/sdk/launch/agent/agent.py

```diff
@@ -227,22 +267,15 @@ def loop(self) -> None:
                                     f"Error running job: {traceback.format_exc()}"
```

Comment:
What happens if a thread has an exception? Does this exception capture it and run `ack_run_queue_item`?